3.workload.md

```diff
@@ -0,0 +1,378 @@
+
+### Spec
+
+The specification section contains a reference to the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `DefinitionRef` | [`DefinitionRef`](#DefinitionRef) | Y | | Workload schema reference. |
+
+
+#### DefinitionRef
+
+DefinitionRef defines where to find the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The workload schema's name. Must be formatted in a way that can uniquely identify a specific version of the definition  |
+
+The workload definition itself __MUST__ contain information that can be used to locate it directly. We will use that
+ information in the [Component](4.component.md) to refer to the workload schema instead of using the name of the workload definition. 
+
+We recommend that the `name` here follows the format described in [the Group, Version, Kind section of part 2](2.overview_and_terminology.md).
+
+We also recommend that the `name` of the workload (in the metadata section) to include the Group and Kind of the schema.
+
+## Workload Kinds
+
+There are three kinds of workloads:
+
+__Core workloads__ are the kind of workloads that every OAM runtime MUST implement according to spec definition
+
+__Standard workloads__ are the kind of workloads that a OAM runtime can choose to implement. However, the
+ implementation MUST strictly adhere to the schematics defined in the spec 
+
+__Extended workloads__ OAM platforms are free to define extra workload kinds and  
+
+### Core Workload Kind
+
+A core workload MUST be in the `core.oam.dev` group.
+
+A core workload MUST all be supported according to the definitions in this section by any runtime implementation of
+ this specification.
+
+Currently, all core workloads are container-based, and assume that an implementation is capable of executing an OCI or Docker image as a container, and are capable of working with OCI registries.
+We may introduce non container-based workloads, such as VM-based, in the future. We can also promote standard workloads to core if there is a consensus in the community.
+
+### Standard Workload Kind
+
+A standard workload MUST be in the `standard.oam.dev` group.
+
+A standard workload MUST be implemented according to the definitions in this section if any runtime chooses to implement it.
+ 
+We do not have any standard workload defined in the current spec. We may introduce new workloads or promote extended workloads to standard if there is a consensus in the community.
+ 
+### Extended Workload Kind
+An extended workload MUST be in the `extended.oam.dev` group.
+
+Extended workload kind are _per runtime_, meaning that each runtime may define its own extended workloads schema beyond this specification. 
+
+## Core Workload
+The following core workload kinds are defined by this specification
+
+### Container Workload
```

Comment:
I strongly suggest to move spec of core workloads (the whole section below) to a separate directory (e.g. `spec/core_workloads/container_workload.md` etc), and we will have more core workloads in the future.

My opinion is they are not part of OAM spec, they just allowed to stay in OAM spec repo because they are "core".

Reply:
I thought about that too, didn't know where to put them.

Reply:
Let's do `spec/core_workloads/container_workload.md`. /cc @negz

Reply:
Happy for it to move out of this doc. I suggest calling it a "_containerised_ workload" rather than a "container workload" though.

Reply:
`containerized` or `containerised`?

Reply:
This is very hard to say an an Australian, but since software tends to use US english it should probably be containeri**z**ed. 😢